jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/ResourceListing.java

```diff
@@ -200,57 +210,125 @@ public static String getAsHTML(Resource resource, String base, boolean parent, S
 
         DateFormat dfmt = DateFormat.getDateTimeInstance(DateFormat.MEDIUM, DateFormat.MEDIUM);
 
-        for (Resource item : listing)
+        for (Path item : listing)
         {
-            // TODO this feels fragile, as collections probably should not return a Path here
-            //      and even if they do, it might not be named correctly
-            String name = item.getPath().toFile().getName();
+            Path fileName = item.getFileName();
+            if (fileName == null)
+            {
+                continue; // skip
+            }
+
+            String name = fileName.toString();
             if (StringUtil.isBlank(name))
-                continue;
+            {
+                return null;
+            }
 
-            if (item.isDirectory() && !name.endsWith("/"))
+            if (Files.isDirectory(item))
+            {
                 name += URIUtil.SLASH;
+            }
 
             // Name
             buf.append("<tr><td class=\"name\"><a href=\"");
-            // TODO should this be a relative link?
-            String path = URIUtil.addEncodedPaths(encodedBase, URIUtil.encodePath(name));
-            buf.append(path);
+
```

Comment:
You have deleted my TODO?  Do you know why we are using absolute rather than relative links here?

Reply:
This is how its has been since Jetty 7.x thru to Jetty 11.x.

Original change in our repo (for Jetty 7.0.0):

https://github.com/eclipse/jetty.project/commit/456a00161b47b8a3a13fc253c3a94ccacd69ea3b#diff-42810570fa3d8b562ad62d94a4ae857667aeb10b17f4ef9e93af43411af3d0aaR457

Current Jetty 11.0.x HEAD:

https://github.com/eclipse/jetty.project/blob/jetty-11.0.x/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java#L694

It's survived multiple updates and fixes since it was introduced.

To change from absolute links to relative links is possible, but out of scope for this PR. (don't want to evolve this to an omnibus PR again)
I'll reintroduce the TODO, but change the text to be more clear.